src/System.Runtime.Serialization.Xml/tests/Performance/PerformanceTestsCommon.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Xunit.Performance;
+using System.Xml;
+using SerializationTypes;
+
+namespace System.Runtime.Serialization
+{
+    #region Performance tests configuration types
+
+    public enum SerializerType
+    {
+        DataContractSerializer,
+        DataContractJsonSerializer,
+        XmlSerializer,
+        JsonNet
+    }
+
+    public enum TestType
+    {
+        SimpleType,
+        ISerializable,
+        Dictionary,
+        ListOfInt,
+        String,
+        ByteArray,
+        XmlElement
+    }
+
+    public interface IPerfTestSerializer
+    {
+        void Init(object obj);
+        void Serialize(object obj, Stream s);
+        void Deserialize(Stream s);
+    }
+
+    public abstract class SerializerFactory
+    {
+        public abstract IPerfTestSerializer GetSerializer(SerializerType serializerType);
+    }
+
+    #endregion
+
+    public class PerformanceTestCommon
+    {
+
+        #region Performance test configuration
+
+        public static IEnumerable<object[]> PerformanceMemberData(SerializerType serializerType)
+        {
+            yield return new object[] { 100, TestType.ByteArray, 1024, serializerType };
+            yield return new object[] { 5, TestType.ByteArray, 1024 * 1024, serializerType };
```

Comment:
1. It's hard to tell what each element is for. 2. Can we eliminate serializerType as it's always the same? 3. We should encapsulate the object creation logic along with other test settings (not necessarily here) and eliminate TestType.